src/main/kotlin/de/cyface/collector/storage/exception/ContentRangeNotMatchingFileSize.kt

```diff
@@ -26,4 +28,14 @@ package de.cyface.collector.storage.exception
  * @version 1.0.1
  * @constructor Provide an explanation for the error via the `format` parameter.
  */
-class ContentRangeNotMatchingFileSize(format: String) : Exception(format)
+class ContentRangeNotMatchingFileSize(format: String) : Exception(format) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is ContentRangeNotMatchingFileSize) return false
+        return this.message == other.message
+    }
+
+    override fun hashCode(): Int {
+        return Objects.hash(javaClass.hashCode(), message)
+    }
+}
```

Comment:
A data class is for passing data structures around easily. That does not satisfy my notion of an Exception. It would technically work, but a data class for me is similar to a struct in Swift and with a Swift-Struct it would not work, as that is passed around by value and not capable of inheriting from other classes. Inheritance should not be a way to avoid a few lines of code (use delegation for that). If something inherits from something else ,it should actually have an 'is-a' relationship.

Someone else posed that question on Stackoverflow: https://stackoverflow.com/questions/60226941/kotlin-custom-exception-as-data-class